server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/DeleteConditionAction.java

```
@@ -22,36 +22,52 @@
-import org.sonarqube.ws.client.qualitygate.QualityGatesWsParameters;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.server.user.UserSession;
+
+import static org.sonar.db.permission.OrganizationPermission.ADMINISTER_QUALITY_GATES;
+import static org.sonarqube.ws.client.qualitygate.QualityGatesWsParameters.PARAM_ID;
 
 public class DeleteConditionAction implements QualityGatesWsAction {
 
-  private final QualityGates qualityGates;
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final QualityGatesWsSupport wsSupport;
 
-  public DeleteConditionAction(QualityGates qualityGates) {
-    this.qualityGates = qualityGates;
+  public DeleteConditionAction(UserSession userSession, DbClient dbClient, QualityGatesWsSupport wsSupport) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+    this.wsSupport = wsSupport;
   }
 
   @Override
   public void define(WebService.NewController controller) {
     WebService.NewAction createCondition = controller.createAction("delete_condition")
-      .setDescription("Delete a condition from a quality gate. Require Administer Quality Gates permission")
+      .setDescription("Delete a condition from a quality gate.<br>" +
+        "Requires the 'Administer Quality Gates' permission")
       .setPost(true)
       .setSince("4.3")
       .setHandler(this);
 
     createCondition
-      .createParam(QualityGatesWsParameters.PARAM_ID)
+      .createParam(PARAM_ID)
       .setRequired(true)
       .setDescription("Condition ID")
       .setExampleValue("2");
   }
 
   @Override
   public void handle(Request request, Response response) {
-    qualityGates.deleteCondition(QualityGatesWs.parseId(request, QualityGatesWsParameters.PARAM_ID));
-    response.noContent();
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      long conditionId = request.mandatoryParamAsLong(PARAM_ID);
```

Should be done before opening a db session - for to fail fast.